source/fe/fe_enriched.cc

```diff
@@ -1046,6 +1048,443 @@ FE_Enriched<dim, spacedim>::InternalData::get_fe_output_object(
+          // the actual predicate id.
+          //
+          // When the grid is refined, material id is inherited to
+          // children cells. So, the cellwise_color_predicate_map stays
+          // relevant.
+          cell->set_material_id(map_index);
+          std::set<unsigned int> color_list;
+
+          // loop through active predicates for the cell and insert map.
+          // Eg: if the cell with material id 100 has active
+          // predicates 4 (color = 1) and 5 (color = 2), the map will insert
+          // pairs (1, 4) and (2, 5) at key 100 (i.e unique id of cell is
+          // mapped with a map which associates color with predicate id)
+          // Note that color list for the cell would be {1,2}.
+          for (unsigned int i = 0; i < predicates.size(); ++i)
+            {
+              if (predicates[i](cell))
+                {
+                  /*
+                   * create a pair predicate color and predicate id and add it
+                   * to a map associated with each enriched cell
+                   */
+                  auto ret = cellwise_color_predicate_map[map_index].insert(
+                    std::pair<unsigned int, unsigned int>(predicate_colors[i],
+                                                          i));
+
+                  AssertThrow(
+                    ret.second == 1,
+                    ExcMessage("Only one enrichment function per color"));
+
+                  color_list.insert(predicate_colors[i]);
+                }
+            }
+
+
+          /*
+           * check if color combination is already added.
+           * If already added, set the active FE index based on
+           * its index in the fe_sets. If the combination doesn't
+           * exist, add the set to fe_sets and once again set the
+           * active FE index as last index in fe_sets.
+           *
+           * Eg: if cell has color list {1,2} associated and
+           * fe_sets = { {}, {1}, {2} } for now, we need to add a new set {1,2}
+           * to fe_sets and a new active FE index 3 because 0 to 2 FE indices
+           * are already taken by existing sets in fe_sets.
+           */
+          bool found = false;
+          if (!color_list.empty())
+            {
+              for (unsigned int j = 0; j < fe_sets.size(); ++j)
+                {
+                  if (fe_sets[j] == color_list)
+                    {
+                      found = true;
+                      cell->set_active_fe_index(j);
+                      break;
+                    }
+                }
+
+              if (!found)
+                {
+                  fe_sets.push_back(color_list);
+                  cell->set_active_fe_index(fe_sets.size() - 1);
+                }
+            }
+          ++map_index; // map_index should be unique to cells
+        }
+
+
+      /*
+       * Treat interface between enriched cells specially
+       * Since the function
```

Comment:
reword it something like:

Until https://github.com/dealii/dealii/issues/1496 is resolved, each time we building constraints at the interface
between two different FE_Enriched, we look for the least dominating FE via hp::FECollection< dim, spacedim >::find_least_face_dominating_fe(). If we don't take further actions, we may find a dominating FE that is too
restrictive, i.e. enriched FE consisting of only....